### PR TITLE
Fix CPM scenario

### PR DESCRIPTION
**Pull request by dimityrmirchev**
**What this PR does / why we need it**:
This PR fixes the control plane migration scenario by preserving the deployed shoot resources.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```bugfix operator
Shoot resources are now preserved during control plane migration.
```

### CODE REVIEW

File: pkg/controller/lifecycle/actuator.go [hunk 1]
```diff
@@ -234,6 +234,11 @@ func (a *actuator) Restore(ctx context.Context, ex *extensionsv1alpha1.Extension
 
 // Migrate the Extension resource.
 func (a *actuator) Migrate(ctx context.Context, ex *extensionsv1alpha1.Extension) error {
+	// Keep objects for shoot managed resources so that they are not deleted from the shoot during the migration
+	if err := managedresources.SetKeepObjects(ctx, a.client, ex.GetNamespace(), constants.ManagedResourceNamesShoot, true); err != nil {
```
Comment (vpnachev) [inline pkg/controller/lifecycle/actuator.go:238]: Should you unset this in `Restrore`?
I am thinking what will happen to a shoot that have been migrated and then disable the extension - will the oidc resources be cleaned-up?

Comment (dimityrmirchev) [reply pkg/controller/lifecycle/actuator.go:238]: On `Restore` the `Reconcile` function is called. In it there is a call to `managedresources.CreateForShoot` with a passed paramerer `false` for `KeepObjects` [ref](https://github.com/gardener/gardener-extension-shoot-oidc-service/blob/4d0f725befb35553a47161cadb068f938206480e/pkg/controller/lifecycle/actuator.go#L162). A new set of `managedresources` will be created that will "adopt" the resources in the shoot.

Comment (vpnachev) [reply pkg/controller/lifecycle/actuator.go:238]: Thanks!

**Review by vpnachev — APPROVED** [review]
/lgtm

**Pull request merged**